### PR TITLE
Fix rendering of key-value rules in API reference

### PR DESCRIPTION
**Pull request by htdvisser**
#### Summary

This fixes rendering of key-value validation rules, which were previously not rendered.

Refs https://github.com/TheThingsIndustries/lorawan-stack-support/issues/263

#### Screenshots

Screenshot 2020-12-28 at 14 48 46

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Run Locally: Verified that the docs build using `make server`, posted screenshots, verified external links.
- [ ] New Features Marked: Documentation for new features is marked using the `new-in-version` shortcode, according to the guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [ ] Style Guidelines: Documentation obeys style guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [x] Commits: Commit messages follow guidelines in [CONTRIBUTING](CONTRIBUTING.md), there are no fixup commits left.

### CODE REVIEW

File: doc/themes/the-things-stack/layouts/shortcodes/proto/message.html [hunk 1]
```diff
@@ -40,6 +40,16 @@ <h3 id="message:{{ .name }}">Message <code>{{ .name }}</code></h3>
             <p><code>{{ $rule }}</code>{{ if not (eq $value true)}}: <code>{{ $value }}</code>{{ end }}</p>
           {{- end }}
         {{ end }}
+        {{- if .map_key }}
+          {{- range $rule, $value := .map_key.rules }}
+            <p><code>{{ $rule }}</code> (key){{ if not (eq $value true)}}: <code>{{ $value }}</code>{{ end }}</p>
```
Comment (kschiffer) [inline doc/themes/the-things-stack/layouts/shortcodes/proto/message.html:45]:
```suggestion
            <p>Key: <code>{{ $rule }}</code>{{ if not (eq $value true)}}, value: <code>{{ $value }}</code>{{ end }}</p>
```

File: doc/themes/the-things-stack/layouts/shortcodes/proto/message.html [hunk 2]
```diff
@@ -40,6 +40,16 @@ <h3 id="message:{{ .name }}">Message <code>{{ .name }}</code></h3>
...
+          {{- end }}
+        {{ end }}
+        {{- if .map_value }}
+          {{- range $rule, $value := .map_value.rules }}
+            <p><code>{{ $rule }}</code> (value){{ if not (eq $value true)}}: <code>{{ $value }}</code>{{ end }}</p>
```
Comment (kschiffer) [inline doc/themes/the-things-stack/layouts/shortcodes/proto/message.html:50]:
```suggestion
            <p>Key: <code>{{ $rule }}</code>{{ if not (eq $value true)}}, value: <code>{{ $value }}</code>{{ end }}</p>
```

Comment (kschiffer) [reply doc/themes/the-things-stack/layouts/shortcodes/proto/message.html:50]: Not sure on this one. Shouldn't this be `key` and `value` too?

**Review by kschiffer — APPROVED** [review]

File: doc/themes/the-things-stack/layouts/shortcodes/proto/message.html (hunk 1 above)
Comment (htdvisser) [reply doc/themes/the-things-stack/layouts/shortcodes/proto/message.html:45]: That would render something like

> Key: `max_length`, value: `42`. Don't think that's what we want.

Comment (kschiffer) [reply doc/themes/the-things-stack/layouts/shortcodes/proto/message.html:45]: The (key) in brackets is weird too :shrug: . I guess we can rather remove it altogether.

**Review by pierrephz — APPROVED** [review]

**Pull request merged**